Calculator.py

```diff
@@ -0,0 +1,76 @@
+class math_funcs(object):
+    """This class include special math funcs"""
+
+    @staticmethod
+    def factorial(num: str):
+        """Doing factorial math function"""
+        fact = 1
+        for n in range(1, int(num) + 1):
+            fact = fact * n
+        return fact
+
+    @staticmethod
+    def average(num1: int, num2: int):
+        """"Doing average between two numbers"""
+        avg = (num1 + num2) / 2
+        return avg
+
+
+class Calculator(object):
+    """The class defines calculator with operations order"""
+
+    def __init__(self, list_of_exercise: list):
+        """
+        Defines five different dictionaries according specific math orders,
+        also getting a list that contains split string that represents math exercise.
+        """
+        self.list_of_exercise = list_of_exercise
+        self.dict_of_operations_volume_5 = {"&": lambda num1, num2: min(num1, num2),
+                                            "$": lambda num1, num2: max(num1, num2),
+                                            "@": lambda num1, num2: math_funcs.average(num1, num2)}
+        self.dict_of_operations_volume_4 = {"!": lambda num: math_funcs.factorial(num),
+                                            "%": lambda num1, num2: num1 % num2}
+        self.dict_of_operations_volume_3 = {"^": lambda num1, num2: num1 ** num2}
+        self.dict_of_operations_volume_2 = {"*": lambda num1, num2: num1 * num2,
+                                            "/": lambda num1, num2: num1 / num2}
+        self.dict_of_operations_volume_1 = {"+": lambda num1, num2: num1 + num2,
+                                            "-": lambda num1, num2: num1 - num2}
+        self.list_of_dicts = [self.dict_of_operations_volume_5, self.dict_of_operations_volume_4,
+                              self.dict_of_operations_volume_3, self.dict_of_operations_volume_2,
+                              self.dict_of_operations_volume_1]
+
+    def change_list(self, result, index: int):
+        """The function changing list values after calculate"""
+        self.list_of_exercise[index] = result
+        self.list_of_exercise.pop(index + 1)
+        self.list_of_exercise.pop(index - 1)
+
+    def scan_and_calculate(self):
```

Comment:
the function must do just one thing!
you violate the single responsibility rule!
split the function to 2 functions

Reply:
The function doesn't scanning, just calculating, it's for better understanding for you..

Reply:
than change your doc and function name